tests/phpunit/integration/Core/Authentication/Has_Multiple_AdminsTest.php

```diff
@@ -34,19 +35,120 @@ public function set_up() {
-		$this->assertFalse( $has_multiple_admins->get(), 'Should return false when there are no admin users' );
+		$this->assertFalse( $has_multiple_admins->get(), 'Should return false when there is only one admin' );
 
 		// Create a second admin, the first one (with ID=1) we already have.
 		$this->factory()->user->create( array( 'role' => 'administrator' ) );
 
-		// Delete the transient and check that we have multiple admins now.
-		$this->transients->delete( Has_Multiple_Admins::OPTION );
+		// Check that we have multiple admins now.
+		$this->assertTrue( $has_multiple_admins->get(), 'Should return true when there are two admin users' );
+		$this->assertEquals( 2, $this->transients->get( Has_Multiple_Admins::OPTION ), 'Transient should be updated to two admins' );
+	}
+
+	public function test_get__on_delete_user() {
+		$has_multiple_admins = new Has_Multiple_Admins( $this->transients );
+		$has_multiple_admins->register();
+
+		// Verify we have only one admin user to start with.
+		$this->assertCount( 1, get_users( array( 'role' => 'administrator' ) ), 'There should be one admin user to begin with' );
+
+		// Create a second admin user and one non-admin user.
+		$admin_id_2   = $this->factory()->user->create( array( 'role' => 'administrator' ) );
+		$non_admin_id = $this->factory()->user->create( array( 'role' => 'editor' ) );
+
+		// Now we have multiple admins.
+		$this->assertTrue( $has_multiple_admins->get(), 'Should return true when there are multiple admin users' );
+
+		// Delete a non-admin user, should not affect the transient.
+		wp_delete_user( $non_admin_id );
+		$this->assertTrue( $has_multiple_admins->get(), 'Should still return true when a non-admin user is deleted' );
+
+		// Delete one of the admin users, should clear the transient and reflect the change.
+		wp_delete_user( $admin_id_2 );
+		$this->assertFalse( $has_multiple_admins->get(), 'Should return false when there is only one admin user left' );
+	}
+
+	public function test_get__on_set_role() {
+		$has_multiple_admins = new Has_Multiple_Admins( $this->transients );
+		$has_multiple_admins->register();
+
+		// Verify we have only one admin user to start with.
+		$this->assertCount( 1, get_users( array( 'role' => 'administrator' ) ), 'There should be one admin user to begin with' );
+
+		// Create a second admin user and one non-admin user.
+		$admin_id_2   = $this->factory()->user->create( array( 'role' => 'administrator' ) );
+		$non_admin_id = $this->factory()->user->create( array( 'role' => 'editor' ) );
+
+		// Now we have multiple admins.
+		$this->assertTrue( $has_multiple_admins->get(), 'Should return true when there are multiple admin users' );
+		$this->assertEquals( 2, $this->transients->get( Has_Multiple_Admins::OPTION ), 'Transient should be updated to two admins' );
+
+		// Change a non-admin user to another non-admin role, should not affect the transient.
+		$user = new WP_User( $non_admin_id );
```

Comment:
We can clean this up a little by getting the user instances from the factory rather than just IDs, and we can of course get the IDs from the instances.